src/ActionServer_FJT.c

```diff
@@ -482,6 +482,142 @@ void Ros_ActionServer_FJT_ProcessFeedback()
+{
+    if (goal_joint_tolerances == NULL || joint_names == NULL || posTolerances == NULL)
+    {
+        Ros_Debug_BroadcastMsg("%s: one or more inputs NULL, abort", __func__);
+        return -1;
+    }
+
+    //always configure defaults
+    for (int i = 0; i < posTolerances_len; ++i)
+        posTolerances[i] = DEFAULT_FJT_GOAL_POSITION_TOLERANCE;
+
+    //if caller hasn't passed any JointTolerances, set all entries to default
+    if (goal_joint_tolerances->size == 0)
+    {
+        Ros_Debug_BroadcastMsg("%s: no joint tolerances specified, returning %d defaults",
+            __func__, posTolerances_len);
+        return OK;
+    }
+
+    //it's OK if we are passed more JointTolerance instances than we have joints,
+    //as technically a client could send duplicate tolerance specs (ie: multiple
+    //JointTolerance instances referring to the same joint).
+    //Last-setting wins, so we can just parse all of them.
+
+    //it's NOT ok to have less elements in the output array than the number of
+    //known joint names: maximum nr of tolerances we can store is equal to the
+    //number of joints this controller is configured with, so we must have at
+    //least that number of elements in the output array
+    if (posTolerances_len < joint_names->size)
+    {
+        Ros_Debug_BroadcastMsg("%s: pos tolerance output array too small: %d < %d (nr of joints)",
+            __func__, posTolerances_len, joint_names->size);
+        return -2;
+    }
+
+    //this:
+    // - loops over all JointTolerance instances
+    // - checks which joint they reference (based on the joint name)
+    // - looks up joint index of that joint in the joint_names array
+    // - stores the position tolerance value in the output array at the correct index for that joint
+    for (size_t jtol_idx = 0; jtol_idx < goal_joint_tolerances->size; jtol_idx +=1)
+    {
+        rosidl_runtime_c__String selected_tolerance_name = goal_joint_tolerances->data[jtol_idx].name;
+        Ros_Debug_BroadcastMsg("%s: parsing JointTolerance for '%s': pos: %f", __func__,
+            selected_tolerance_name.data, goal_joint_tolerances->data[jtol_idx].position);
+
+        size_t ptol_idx = 0;
+        for (; ptol_idx < joint_names->size; ptol_idx += 1)
+        {
+            rosidl_runtime_c__String selected_position_name = joint_names->data[ptol_idx];
+            if (rosidl_runtime_c__String__are_equal(&selected_position_name, &selected_tolerance_name))
+            {
+                Ros_Debug_BroadcastMsg("%s: mapping '%s' (at %d) to internal index %d",
+                    __func__, selected_position_name.data, jtol_idx, ptol_idx);
+                posTolerances[ptol_idx] = goal_joint_tolerances->data[jtol_idx].position;
+                break;
+            }
+        }
+
+        //couldn't find joint
+        if (ptol_idx == joint_names->size)
+        {
+            //TODO(gavanderhoorn): make this a fatal error?
```

Comment:
I assume (hope) that you mean fatal to the trajectory request (return not OK) and not a fatal assertion.

Reply:
Yes, even more 'local' perhaps: fatal as in: exit this function with an appropriate return code.